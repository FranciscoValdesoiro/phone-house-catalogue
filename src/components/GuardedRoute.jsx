import React from 'react';
import { Route, Redirect } from "react-router-dom";

const GuardedRoute = ({auth, component: Component, path}) => {
  return (
    <Route path={path} render={(props) => (
        
        auth ? <Component {...props} />
        : <Redirect to='/' />
    
    )} />
  )           
}

export default GuardedRoute;
