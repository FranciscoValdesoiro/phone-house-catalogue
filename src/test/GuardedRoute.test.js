import React from 'react';
import GuardedRoute from '../components/GuardedRoute.jsx'
import { mount } from 'enzyme'

import '@testing-library/jest-dom'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'


describe('GuardedRoute', () => {
 
const history = createMemoryHistory()
 
  it('Should render the GuardedRoute component', () => {

    const wrapper = mount( <Router history={history}>
        <GuardedRoute  />
      </Router>)
    expect(wrapper).toMatchSnapshot()
  })

});