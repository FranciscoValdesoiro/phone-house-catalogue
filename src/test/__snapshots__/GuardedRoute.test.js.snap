// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuardedRoute Should render the GuardedRoute component 1`] = `""`;
